mm2src/mm2_main/src/lp_swap/swap_v2_common.rs

```diff
@@ -371,3 +373,99 @@ pub(super) async fn swap_kickstart_handler<
+                MakerSwapStateMachine<UtxoStandardCoin, UtxoStandardCoin>,
+                UtxoStandardCoin,
+                UtxoStandardCoin,
+            >(swap_repr, storage, uuid, m, t)
+            .await
+        },
+        (MmCoinEnum::EthCoin(m), MmCoinEnum::EthCoin(t)) => {
+            swap_kickstart_handler::<MakerSwapStateMachine<EthCoin, EthCoin>, EthCoin, EthCoin>(
+                swap_repr, storage, uuid, m, t,
+            )
+            .await
+        },
+        (MmCoinEnum::UtxoCoin(m), MmCoinEnum::EthCoin(t)) => {
+            swap_kickstart_handler::<MakerSwapStateMachine<UtxoStandardCoin, EthCoin>, UtxoStandardCoin, EthCoin>(
+                swap_repr, storage, uuid, m, t,
+            )
+            .await
+        },
+        (MmCoinEnum::EthCoin(m), MmCoinEnum::UtxoCoin(t)) => {
+            swap_kickstart_handler::<MakerSwapStateMachine<EthCoin, UtxoStandardCoin>, EthCoin, UtxoStandardCoin>(
+                swap_repr, storage, uuid, m, t,
+            )
+            .await
+        },
+        _ => {
+            error!(
+                "V2 swaps are not currently supported for {}/{} pair",
+                swap_repr.maker_coin(),
+                swap_repr.taker_coin()
+            );
+        },
+    };
+}
+
+pub(super) async fn swap_kickstart_handler_for_taker(
+    ctx: MmArc,
+    swap_repr: TakerSwapDbRepr,
+    storage: TakerSwapStorage,
+    uuid: Uuid,
+) {
+    let (maker_coin, taker_coin) = match swap_kickstart_coins(&ctx, &swap_repr, &uuid).await {
+        Some(coins) => coins,
+        None => return,
+    };
+
+    match (maker_coin, taker_coin) {
+        (MmCoinEnum::UtxoCoin(m), MmCoinEnum::UtxoCoin(t)) => {
+            swap_kickstart_handler::<
+                TakerSwapStateMachine<UtxoStandardCoin, UtxoStandardCoin>,
+                UtxoStandardCoin,
+                UtxoStandardCoin,
+            >(swap_repr, storage, uuid, m, t)
+            .await
+        },
+        (MmCoinEnum::EthCoin(m), MmCoinEnum::EthCoin(t)) => {
+            swap_kickstart_handler::<TakerSwapStateMachine<EthCoin, EthCoin>, EthCoin, EthCoin>(
+                swap_repr, storage, uuid, m, t,
+            )
+            .await
+        },
+        (MmCoinEnum::UtxoCoin(m), MmCoinEnum::EthCoin(t)) => {
+            swap_kickstart_handler::<TakerSwapStateMachine<UtxoStandardCoin, EthCoin>, UtxoStandardCoin, EthCoin>(
+                swap_repr, storage, uuid, m, t,
+            )
+            .await
+        },
+        (MmCoinEnum::EthCoin(m), MmCoinEnum::UtxoCoin(t)) => {
+            swap_kickstart_handler::<TakerSwapStateMachine<EthCoin, UtxoStandardCoin>, EthCoin, UtxoStandardCoin>(
+                swap_repr, storage, uuid, m, t,
+            )
+            .await
+        },
+        _ => {
+            error!(
+                "V2 swaps are not currently supported for {}/{} pair",
+                swap_repr.maker_coin(),
+                swap_repr.taker_coin()
+            );
+        },
+    };
```

Comment:
I think you can omit the type names for every match arm..e.g

```rust
swap_kickstart_handler::<TakerSwapStateMachine<_, _>, _, _>(swap_repr, storage, uuid, m, t).await
```

instead of 

```rust
 swap_kickstart_handler::<
                TakerSwapStateMachine<UtxoStandardCoin, UtxoStandardCoin>,
                UtxoStandardCoin,
                UtxoStandardCoin,
            >(swap_repr, storage, uuid, m, t)
            .await
```

Reply:
Looks better https://github.com/KomodoPlatform/komodo-defi-framework/pull/2300/commits/1065997604e3d1f250b80d384f644f3e60107d2e